src/validators/shapes.rs

```diff
@@ -17,7 +19,54 @@ pub fn validate(gtfs: &gtfs_structures::Gtfs) -> Vec<Issue> {
             Issue::new(Severity::Error, IssueType::InvalidCoordinates, id)
                 .object_type(gtfs_structures::ObjectType::Shape)
         });
-    missing_coord.chain(valid).collect()
+    let invalid_shape_id = gtfs
+        .trips
+        .iter()
+        .map(|(_id, trip)| create_invalid_shape_id_issue(trip, gtfs))
+        .filter_map(|o| o);
+
+    let used_shape_id: HashSet<String> = gtfs
+        .trips
+        .iter()
+        .map(|(_trip_id, trip)| trip.shape_id.to_owned())
```

Comment:
you can just use `filter_map` here and remove the next filter_map :stuck_out_tongue_winking_eye: